Terminal.Gui/Input/Keyboard.cs

```diff
@@ -0,0 +1,635 @@
+	/// </summary>
+	public bool Handled { get; set; } = false;
+
+	/// <summary>
+	/// Symbolic definition for the key.
+	/// </summary>
+	public Key Key { get; set; }
+
+	/// <summary>
+	/// The key value as a Rune. This is the actual value of the key pressed, and is independent of the modifiers.
+	/// </summary>
+	/// <remarks>
+	/// If the key pressed is a letter (a-z or A-Z), this will be the upper or lower case letter depending on whether the shift key is pressed.
+	/// If the key is outside of the <see cref="Key.CharMask"/> range, this will be <see langword="default"/>.
+	/// </remarks>
+	public Rune AsRune => ToRune (Key);
+
+	/// <summary>
+	/// Converts a <see cref="Key"/> to a <see cref="Rune"/>.
+	/// </summary>
+	/// <remarks>
+	/// If the key is a letter (a-z or A-Z), this will be the upper or lower case letter depending on whether the shift key is pressed.
+	/// If the key is outside of the <see cref="Key.CharMask"/> range, this will be <see langword="default"/>.
+	/// </remarks>
+	/// <param name="key"></param>
+	/// <returns>The key converted to a rune. <see langword="default"/> if conversion is not possible.</returns>
+	public static Rune ToRune (Key key)
+	{
+		if (key is Key.Null or Key.SpecialMask || key.HasFlag (Key.CtrlMask) || key.HasFlag (Key.AltMask)) {
+			return default;
+		}
+
+		// Extract the base key (removing modifier flags)
+		Key baseKey = key & ~Key.CtrlMask & ~Key.AltMask & ~Key.ShiftMask;
+
+		switch (baseKey) {
+		case >= Key.A and <= Key.Z when !key.HasFlag (Key.ShiftMask):
+			return new Rune ((char)(baseKey + 32));
+		case >= Key.A and <= Key.Z:
+			return new Rune ((char)baseKey);
+		case >= Key.Null and < Key.A:
+			return new Rune ((char)baseKey);
+		}
+
+		if (Enum.IsDefined (typeof (Key), baseKey)) {
+			return default;
+		}
+
+		return new Rune ((char)baseKey);
+	}
+
+	/// <summary>
+	/// Gets a value indicating whether the Shift key was pressed.
+	/// </summary>
+	/// <value><see langword="true"/> if is shift; otherwise, <see langword="false"/>.</value>
+	public bool IsShift => (Key & Key.ShiftMask) != 0;
+
+	/// <summary>
+	/// Gets a value indicating whether the Alt key was pressed (real or synthesized)
+	/// </summary>
+	/// <value><see langword="true"/> if is alternate; otherwise, <see langword="false"/>.</value>
+	public bool IsAlt => (Key & Key.AltMask) != 0;
+
+	/// <summary>
+	/// Gets a value indicating whether the Ctrl key was pressed.
+	/// </summary>
+	/// <value><see langword="true"/> if is ctrl; otherwise, <see langword="false"/>.</value>
+	public bool IsCtrl => (Key & Key.CtrlMask) != 0;
+
+	/// <summary>
+	/// Gets a value indicating whether the key is a letter (a-z or A-Z). This is independent of the shift key.
+	/// </summary>	
+	public bool IsAlpha {
+		get {
+			if (IsAlt || IsCtrl) {
+				return false;
+			}
+			return (Key & Key.CharMask) is >= Key.A and <= Key.Z;
```

Comment:
This is wrong because only will return `true` if it's from `A` to `Z` and not if it's from accented chars like `É` which has the `É+Key.ShiftMask`. It should be like this:

```cs
return char.IsLetterOrDigit ((char)(Key & Key.CharMask));
```

Reply:
Nope. See same but different question from Thomas above.

Reply:
I should rename it to "IsKeyBetweenAandZ". 

I want to fix the need for this though.